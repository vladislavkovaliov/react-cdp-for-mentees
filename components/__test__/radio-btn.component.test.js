import React from 'react';
import { RadioButton } from '../radio-btn.component';
import { shallow } from 'enzyme';

const mockOnChange = jest.fn();

describe('<RadioButton />', () => {
  describe('Shapshot', () => {
    test('should render <RadioButton /> with default option', () => {
      const wrapper = shallow(
        <RadioButton
          value="title"
          name="title"
          id="title"
          onChange={mockOnChange}
          isChecked={true}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });

    test('should render <RadioButton /> without default option', () => {
      const wrapper = shallow(
        <RadioButton
          value="title"
          name="title"
          id="title"
          onChange={mockOnChange}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Behavior', () => {

  });
});
