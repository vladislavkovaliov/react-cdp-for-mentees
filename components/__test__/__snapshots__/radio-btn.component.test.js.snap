// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> Shapshot should render <RadioButton /> with default option 1`] = `
<React.Fragment>
  <input
    checked={true}
    id="title"
    name="title"
    onChange={[MockFunction]}
    type="radio"
    value="title"
  />
  <label
    htmlFor="title"
  />
</React.Fragment>
`;

exports[`<RadioButton /> Shapshot should render <RadioButton /> without default option 1`] = `
<React.Fragment>
  <input
    id="title"
    name="title"
    onChange={[MockFunction]}
    type="radio"
    value="title"
  />
  <label
    htmlFor="title"
  />
</React.Fragment>
`;
